### ISSUE TITLE
About Unsupervised Loss

### ISSUE THREAD

**Issue by DQiaole** [iwbn/flow-supervisor#3]
Hi, thank you for the impressive work!
I have some questions about the usage of $\lambda_{TU}$ in the code. After reading the code, I find the weight of smurf loss are zero default:

https://github.com/iwbn/flow-supervisor/blob/31f50f5697f53b2a0a8a6a47a80af7ee609145ca/raft/semi.py#L546

Is the official model of C+T+ $S^u$ optimized by $L_{FS}$ and $L_{TS}$ solely? I didn't find teacher_smurf_weight here:

https://github.com/iwbn/flow-supervisor/blob/31f50f5697f53b2a0a8a6a47a80af7ee609145ca/train.sh#L9-L16

But I do find it sets a teacher_smurf_weight great than zero on C+T+ $K^u$. And the README.md indicates that ckpts/semi/kitti2/ is trained on C+T+K^u w/o unsup. However, the command in train.sh indicates that --teacher_smurf_weight=0.01 for kitti2. Did I misunderstand something?

Besides, I also find there is no Unsupervised Loss in the pytorch codes (which provides the codes for supervised finetuning C+T+S+K+H+ $Spr^u$/ $K^u$):

https://github.com/iwbn/flow-supervisor/blob/31f50f5697f53b2a0a8a6a47a80af7ee609145ca/pytorch/train_gma.py#L270-L282

Please forgive me for asking so many questions. Also, can you provide codes in pytorch for C+T+ $S^u$/ $K^u$?

Finally, I find a small 'bug', the 'elif' should by 'if' below.

https://github.com/iwbn/flow-supervisor/blob/31f50f5697f53b2a0a8a6a47a80af7ee609145ca/pytorch/wb_data/flow_datasets.py#L14-L18

Thank you very much in advance.

**Comment by iwbn** [comment]
Thank you for your attention to our work! Thanks to your comment, we have fixed some errors in our released code.

### Weight of SMURF loss
**Default settings**
In Table 3 in the paper, we showed that the unsupervised loss is not always helpful for a better semi-supervised accuracy. Thus, for Sintel, we set $\lambda_{TU}=0.0$ as default. Also, for 
| TU   | Sintel Final | KITTI EPE |
| ---- | ------------ | --------- |
| Off  | **2.46**     | 4.69      |
| On   | 2.80         | **3.35**  |

For the ablation study on Sintel, you can edit the option to:
```sh
--census_weight 1. --smooth1_weight 2.5 --smooth2_weight 0. --selfsup_weight 0.0 --smurf_occlusion wang
```
Note that the parameters, we follow the parameters in [SMURF](https://github.com/google-research/google-research/tree/master/smurf) repository; we might be able to find a better hyper-parameters or better unsupervised loss functions for our setting in the future.

**Lambda value**
We have re-checked the released pretrained weights, and we have found an error in the released version.

The setting used fot training is in the `args.yaml` in each pretrained model; and we corrected the `train.sh` and the readme file. See https://github.com/iwbn/flow-supervisor/commit/9e61148779533da22691d03ad90c9940c53a13f8 .


**Spring and pytorch code**
In our paper, we use the Tensorflow implementation of [SMURF](https://github.com/google-research/google-research/tree/master/smurf) for our unsupervised loss, and for pytorch experiments, we did not utilize the TU loss.
Also, for the case where we have a partial label (such as Sintel(labeled) + Spring(unlabeled) ), we found that using unsupervised loss does not help. If you want to attach unsupervised loss to our pytorch framework, you can use an unsupervised loss function implemented in PyTorch (such as [ARFlow](https://github.com/lliuz/ARFlow)).

### Error in the pytorch dataset code
Thank you so much for reporting the typo made during exporting the original code; we fixed it. See https://github.com/iwbn/flow-supervisor/commit/9e61148779533da22691d03ad90c9940c53a13f8

**Comment by DQiaole** [comment]
Thank you for your quick reply and clarification.
